client/src/game/reducers/noteIdentity.ts

```diff
@@ -47,91 +47,88 @@ export function parseIdentity(variant: Variant, keyword: string): CardIdentity {
   return { suitIndex, rank };
 }
 
-function possibilitiesComplement(
-  variant: Variant,
-  possibilities: Array<[number, number]>,
-): Array<[number, number]> {
-  const complement: Array<[number, number]> = [];
-  for (let suitIndex = 0; suitIndex < variant.suits.length; suitIndex++) {
-    for (const rank of variant.ranks) {
-      // Ensure that this possibility is not present in the source list
-      if (
-        !possibilities.some(
-          ([negatedSuitIndex, negatedRank]) =>
-            negatedSuitIndex === suitIndex && negatedRank === rank,
-        )
-      ) {
-        complement.push([suitIndex, rank]);
+function identityMapToArray(cardMap: number[][]): Array<[number, number]> {
+  const rows = cardMap.length;
+  const cols = cardMap[0].length;
+  const possibilities: Array<[number, number]> = [];
+  for (let row = 0; row < rows; ++row) {
+    for (let col = 0; col < cols; ++col) {
+      if (cardMap[row][col]) {
+        possibilities.push([col, row + 1]);
       }
     }
   }
-
-  return complement;
+  return possibilities;
 }
 
 // Examines a single square bracket-enclosed part of a note (i.e. keyword) and returns the set of
 // card possibilities that it declares
 //
 // e.g. the note keyword `red` would return `[[0,1], [0,2], [0,3], [0,4], [0,5]]`
 // and the note keyword `red 3, blue 3` would return `[[0,3], [1,3]]`
+// and the note keyword `r,b,2,3, blue 3` would return `[[0,2], [1,2], [0,3], [1,3]]`
```

Comment:
I find this surprising. It makes more sense to me that `r,b,2,3` means the union of all red cards + all blue cards + all 2's + all 3's. `[r,b] [2,3]` would mean to me what you describe.

Reply:
(The behavior I'm describing is what it did before. )

Reply:
you can still do `[r,b] [2,3]` but this way you don't need the brackets. The OR interpretation doesn't really make sense, IMHO. Who would ever want to mark a card like such?

Reply:
Consider the following scenario in a 3-player game
- Red 2 and Green 2 are on on the stacks
- Cathy has a green 3 on finesse position.
- Bob has three unclued cards.
- Alice clues blue to Cathy touching an already fully known blue 3
- Bob blind-plays red 3 from Finesse Position and realizes it's a 3 bluff.
- Cathy discards.
- Now Bob is sad because green 1 left finesse position. If Alice had just clued him directly on the red 3, then Bob could've bluffed the green 1.
- Bob realizes that if nobody made a mistake, then Alice must not have been able to clue the red 3 directly, therefore he has a red card or a 2 on chop, and a trash red card or a trash 2 in second finesse position.
- Bob writes the note `r,2` on his chop, and it turns into a picture of the red 2
- Bob wishes that this change to note behavior was not made

Reply:
that seems pretty contrived

Reply:
but yes, `r, 3` naturally should represent red or 3.
the fact that it doesn't is just a consequence of your change to remove comma parsing, so it seems like your fault if anything

Reply:
3-player Rainbow game:
- Red 5 and rainbow 3 are on that stacks
- Two copies of rainbow 1 and one copy of rainbow 2 are in the discard pile
- Alice clues red to Cathy
- Bob discharges
- Cathy knows that this is a trash card, so writes the note `[red, 3] [kt]`
- The card morphs into a red 3 much to Cathy's dismay

Reply:
so... what is the consensus here?  r,3= red OR 3?

Reply:
i was gonna wait for timo to respond

Reply:
to my 2nd comment, not my 1st comment

Reply:
> but yes, `r, 3` naturally should represent red or 3.
> the fact that it doesn't is just a consequence of your change to remove comma parsing (maybe?), so it seems like your fault if anything

You mean it's my fault? I don't think I removed comma parsing

Reply:
> so... what is the consensus here? r,3= red OR 3?

That's where I'm at

Reply:
i thought your change was to make `r, b` === `rb`

Reply:
You mean shorthand like `rb3` means `r3,b3`? I never implemented that

Reply:
anyways, back on topic, yes i think whatever system that we come up with, a note of `r, 3` must not transform the card into a red 3

Reply:
iirc timo smartly suggested previous that multiple suits were ANDed to gether, and multiple ranks were ANDed together, and then the suits+ranks are ORed together

Reply:
I don't think that makes sense. A card can't be red AND blue

Reply:
youre right, what suggestion am i thinking of

Reply:
oh i think i was thinking of this https://github.com/Zamiell/hanabi-live/pull/2028#issuecomment-770447023

Reply:
maybe its not relevant since the squishing code isn't implemented yet

Reply:
Ok, I implemented squish notes which should make everyone happy. So now `[b,r][2,3] === rb23` and `r3 !== r,3` but rather `r,3` is red OR 3.  Squish notation pretty much covers all abbreviations, but I left in the old regex as legacy.